app/src/main/java/org/dicekeys/app/openssh/OpenSslHelper.kt

```diff
@@ -0,0 +1,83 @@
+package org.dicekeys.app.openssh
+
+import com.google.common.io.BaseEncoding
+import com.google.common.io.ByteArrayDataOutput
+import com.google.common.io.ByteStreams
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters
+
+object OpenSslHelper {
+    private fun marshalData(out: ByteArrayDataOutput, data: ByteArray) {
+        out.writeInt(data.size) // 32-bit length
+        out.write(data)
+    }
+
+    private fun marshalData(out: ByteArrayDataOutput, str: String): ByteArrayDataOutput {
+        marshalData(out, str.toByteArray(Charsets.US_ASCII))
+        return out
+    }
+
+    private fun marshalPublicKeyEd25519(pubKey: ByteArray): ByteArray {
+        val out = ByteStreams.newDataOutput()
+        marshalData(out, "ssh-ed25519")
+        marshalData(out, pubKey)
+        return out.toByteArray()
+    }
+
+    private fun marshalPrivateKeyEd25519(
+        privateKey: ByteArray,
+        pubKey: ByteArray,
+        comment: String
+    ): ByteArray {
+        val out = ByteStreams.newDataOutput()
+
+        // Per spec this should be a random number just to validate the key if is encrypted
+        // in our case (unencrypted keys) and to also have valid unit tests is hardcoded.
+        // val checksum = Math.random().toInt()
```

Comment:
Confused here.  Can we pass in checksum override as a parameter and, if it's not passed, generate one at random?